core/src/components/card/card.tsx

```diff
@@ -73,48 +67,59 @@ export class TdsCard {
 
   getCardContent = () => (
     <div>
-      {this.headerPlacement === 'above' && (
-        <div class={`card-top ${this.headerPlacement}`}>
-          {this.headerImg && (
-            <img class={`card-top-image`} src={this.headerImg} alt={this.headerImgAlt} />
-          )}{' '}
+      {this.imagePlacement === 'above-header' && (
+        <div class={`card-top ${this.imagePlacement}`}>
+          <div class={hasSlot('card-thumbnail', this.host) ? 'card-thumbnail' : 'no-header-img'}>
+            <slot name="card-thumbnail"></slot>
+          </div>
           <div
-            class={`
-          card-top-header
-          ${!this.headerImg ? 'no-header-img' : ''}
-          ${!this.header || !this.subheader ? 'single-line-header' : ''}
-          `}
+            class={`card-top-header ${!hasSlot('card-thumbail', this.host) ? 'no-header-img' : ''}`}
           >
-            <span class={`card-header`}>{this.header}</span>
-            <span class={`card-subheader`}>{this.subheader}</span>
+            <span class={`card-header`}>
+              {this.header}
+              <slot name="card-header"></slot>
+            </span>
+            <span
+              class={`card-subheader  ${
+                !this.subheader && !hasSlot('card-subheader', this.host) ? 'no-subheader' : ''
+              }`}
```

Comment:
This logic could be split out to a variable to improve readability.
Eg. `const hasSubheader = this.subheader || hasSlot('card-subheader', this.host);`

And then
`class={`card-subheader  ${!hasSubheader ? 'no-subheader' : ''}`

Reply:
Yes! Wasn't sure were to put that evaluation since we want them to be able to do some stuff in runtime, but I guess we could just put it in the render function.

Reply:
Yes I think that's better than not!